src/braket/circuits/moments.py

```diff
@@ -181,6 +182,8 @@ def _add(self, instruction: Instruction, noise_index: int = 0) -> None:
             self._time_all_qubits = time
         elif isinstance(operator, Noise):
             self.add_noise(instruction)
+        elif instruction.target == QubitSet([]):
+            self._global_phase += operator.angle
```

Comment:
if you're assuming this can only happen for a global phase instruction, maybe first `assert isinstance(operator, GPhase)`?

Reply:
You may be able to avoid circular imports by further specifying the import (ie. `braket.circuits.gates` vs `braket.circuits`)